### DISCUSSION TITLE
Diff, Compare and Merge external tool by file extension or type

### DISCUSSION THREAD

**Discussion by BriceDet** [gitextensions/gitextensions#11737] [Ideas]
### Feature description

Good morning,

In order to avoid having external scripts and batches,
I want to be able to configure different external compare/merge/diff tools (and settings) depending on the file type (e.g. .xlsx, .docx, .c, .h, ...)

I didn't find this feature proposal in the backlog. Sorry if I missed it!

### Environment

- GitExtensions version: > 3 
- GIT version: last
- OS version: Windows 10 / 11
- .NET version: -

**Comment by RussKie** [comment]
Could you please elaborate how you expect to achieve this?

**Comment by BriceDet** [comment]
for exemple, all this section
![image](https://github.com/gitextensions/gitextensions/assets/36343321/a988dd92-0f8e-4e81-92d8-328035a145e0)
should have an advanced menu, with specific path/parameter y file type, quietly like in other git client
![image](https://github.com/gitextensions/gitextensions/assets/36343321/65efd026-9b2f-470d-af81-f5b182fe8443)

**Comment by gerhardol** [comment]
You can handle this manually. All difftools configured in Git are available, so you can have multiple difftools.
Similar for merge tools.

![image](https://github.com/gitextensions/gitextensions/assets/6248932/04542b6f-df9e-405c-a582-b5d55ddf634d)

**Comment by drt1** [comment]
How do I manage to get a new entry in this mergetool dropdown?
![image](https://github.com/gitextensions/gitextensions/assets/3314720/d5363c60-a4df-41f8-b106-d913f1d1ef62)

i have added a mergetool globally to git with
`git config --global merge.tool mymergetool`
`git config --global mergetool.mymergetool.cmd <path and arguments>`

It is listed as user-defined when running
`git mergetool --tool-help` 
but it does not show up in the dropdown in Git Extensions.

**Comment by BriceDet** [comment]
Ok @gerhardol
Now I have one of last GitExtention version on my office computer. But.... have no accès to First->Second "sub menu"
May same problem as @drt1

**Reply by gerhardol** [reply]
![image](https://github.com/gitextensions/gitextensions/assets/6248932/b333a6d6-c620-49e1-bfbe-cc40976e9556)

It can take over 10s for Git to generate the list

**Reply by BriceDet** [reply]
Thank you @gerhardol, options were goods :)
I'll take a look on multi diff tool configuration. I think i understand how to do it, but it's not user friendly in the gitExtention GUI (as describe first in my post)
